### PR TITLE
Support any channel types from Notification

### LINKED ISSUE #713: [BUG] New Notification channel types wont be usable and crashes monitor page
**What is the bug?**
When choosing a new Notification channel type that was not supported by Alerting before 2.0, the UI page crashes, since we had the channel types hard-coded. This would crash for SES, SNS, and Microsoft teams.

**What is the expected behavior?**
Update the logic so the channel types are not hard-coded, so we can continuously support the new channel types that the Notification plugin adds.

### PR DESCRIPTION
**Pull request by lezzago**
### Description
Support any channel types from Notification
 
### Issues Resolved
#713 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/CreateTrigger/containers/ConfigureActions/ConfigureActions.js [hunk 1]
```diff
@@ -9,11 +9,6 @@ import { EuiPanel, EuiText, EuiSpacer, EuiLoadingSpinner } from '@elastic/eui';
 import Action from '../../components/Action';
 import ActionEmptyPrompt from '../../components/ActionEmptyPrompt';
 import AddActionButton from '../../components/AddActionButton';
-import {
-  CHANNEL_TYPES,
```
Comment (bowenlan-amzn) [inline public/pages/CreateTrigger/containers/ConfigureActions/ConfigureActions.js:13]: nit: if CHANNEL_TYPES is not used anywhere now, can delete it

**Comment by bowenlan-amzn** [comment]
Why all the cypress tests are failing?

**Comment by lezzago** [comment]
> Why all the cypress tests are failing?

The github CI looks like it needs to be updated. I have tested this locally against the functional test repo and all the CIs passed.

**Review by bowenlan-amzn — DISMISSED** [review]

File: public/services/NotificationService.ts [hunk 2]
```diff
@@ -25,6 +26,18 @@ export default class NotificationService {
...
 
+  getServerFeatures = async (): Promise<Array<String>> => {
+    try {
+      const response = await this.httpClient.get(
+        NODE_API.GET_AVAILABLE_FEATURES
+      );
+      return response.allowed_config_type_list as Array<String>;
+    } catch (error) {
+      console.error('error fetching available features', error);
+      return null;
```
Comment (AWSHurneyt) [inline public/services/NotificationService.ts:37]: Are there any concerns with receiving a null pointer exception somewhere by returning `null` here? Would it be worthwhile to instead return `{}`?

Comment (lezzago) [reply public/services/NotificationService.ts:37]: good point, that would be more robust.

**Review by AWSHurneyt — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]

**Pull request merged**